### PR TITLE
Drop support for Python 3.7.

### PR DESCRIPTION
**Pull request by icemac**

### CODE REVIEW

**Comment by icemac** [comment]
Additionally to the broken MacOS jobs the code for setting the branch protection rules seems to be wrong here.

**Comment by dataflake** [comment]
Looks like this is the first `zope-product` project that has set `with-macos = true` other than Zope itself. Obviously, there's a bug. But is it really necessary to run macOS tests here?

**Comment by dataflake** [comment]
P.S.: @tseaver had a good point recently that running a bunch of tests that aren't really important are a waste of energy. We're continuing to do that in many places because we're not paying for it.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -61,7 +56,11 @@ jobs:
         python -m pip install --upgrade pip
         pip install tox
     - name: Test
+      if: ${{ !startsWith(runner.os, 'Mac') }}
       run: tox -e ${{ matrix.config[1] }}
+    - name: Test (macOS)
+      if: ${{ startsWith(runner.os, 'Mac') }}
+      run: tox -e ${{ matrix.config[1] }}-universal2
```
Comment (davisagli) [inline .github/workflows/tests.yml:63]: I think this causes the failures on macos since this environment isn't defined in tox.ini?

**Review by davisagli — COMMENTED** [review]
Running the tests on macos was added in https://github.com/zopefoundation/Products.ZCatalog/pull/141 because the query plan was sometimes different there.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (dataflake) [reply .github/workflows/tests.yml:63]: Yes it does, I know.

**Comment by icemac** [comment]
Testing on MacOS was added in #141 because of the bug in #140. It seems to behave differently that's why we are testing it.

**Comment by icemac** [comment]
Note: I do not plan to work on fixing this PR.

**Comment by dataflake** [comment]
I will fix it.

**Review by dataflake — APPROVED** [review]

**Comment by dataflake** [comment]
The fix for macOS is the same as already used for Zope itself: C-based packages must be pre-installed with pip because zc.buildout/setuptools messes it all up. This dependency pre-install might be moved from individual packages' meta configuration into the meta templates itself, if I encounter another package like that I'll handle it.

**Comment by icemac** [comment]
Thank you for working on this PR. 😃

**Pull request merged**